### ISSUE TITLE
ENH: grow and shrink methods

### ISSUE THREAD

**Issue by venaturum** [staircase-dev/piso#17]
#### Describe the solution you'd like

specify a length, and one of {"left", "right", "both"} and it grows or shrinks every interval accordingly

**Comment by mesejo** [comment]
I would like to work on this

**Comment by venaturum** [comment]
@mesejo Hey Daniel, this project is still under rapid development, and is a bit unstable.  I'm not sure these methods will be included, and I'm not even sure what these methods should do due to the two different interpretations of an array of intervals: see "with respect to set operations an interval array can be interpreted in one of two ways...." at the bottom of https://piso.readthedocs.io/en/latest/user_guide/intervalsets.html

We could merge intervals or keep them separate before - and after - a shrink or grow operation, and the two options come from the two different interpretations.  

Long story short, I suggest we hold off on this one for now.  If you'd like to contribute to a similar project then I'd suggest looking through [issues in the staircase repo](https://github.com/staircase-dev/staircase/issues).   That project is stable and has a [development guide](https://www.staircase.dev/en/latest/development/index.html)

Cheers, Riley.

**Comment by mesejo** [comment]
Hey Riley! Thanks for answering. I would take a look at the issues, would you like me to help in a particular one?

**Comment by venaturum** [comment]
@mesejo I've added a few more items to the Issue tracker which might interest you if you wanted to contribute to this project
